src/main/java/subway/stations/SubwayStations.java

```diff
@@ -30,6 +30,61 @@ public int getObjectId() {
         }
     }
 
+    public List<Station> dijkstrasAlgorithm(Station source, Station destination, SubwayLines lines) {
+
+        Set<Station> unvisitedStations = new HashSet<>();
+
+        for(Station station : stations){
+            station.distanceFromSource = Integer.MAX_VALUE;
+            station.connections = getConnections(lines, station);
+            unvisitedStations.add(station);
+        }
+        Station currentSource = source;
+        currentSource.distanceFromSource = 0;
+        int shortestDistance;
+
+        while (!(currentSource.connections.contains(destination))) {
+            for(Station station : currentSource.connections){
+                if(unvisitedStations.contains(station)){
+                    station.distanceFromSource = currentSource.distanceFromSource + 1;
+                    station.previous = currentSource;
+                }
+            }
+            unvisitedStations.remove(currentSource);
+            shortestDistance = currentSource.distanceFromSource + 1;
+
+            for (Station station : unvisitedStations) {
+                int thisDistance = station.distanceFromSource;
+                if (thisDistance == shortestDistance) {
+                    currentSource = station;
+                }
+            }
+            for(Station station : unvisitedStations){
+                int thisDistance = station.distanceFromSource;
+                if (thisDistance < shortestDistance) {
+                    shortestDistance = thisDistance;
+                    currentSource = station;
+                }
+            }
+        }
+
+        return getPath(source, destination, currentSource);
+    }
+
+    private List<Station> getPath(Station source, Station destination, Station currentSource) {
+        List<Station> path = new ArrayList<>();
+        path.add(destination);
```

Comment:
destination should be added at the **END** of the path

Reply:
I create the list from the end of the path, and then reverse it

Reply:
I guess i'm confused why you need 3 stations to do this.
You should only use two.